python/cudf/cudf/tests/test_json.py

```diff
@@ -618,3 +618,16 @@ def test_json_nested_lines(data):
     # In the second test-case:
     # Pandas omits "f1" in first row, so we have to enforce a common schema
     assert df.to_arrow().equals(pa.Table.from_pandas(pdf))
+
+
+def test_json_nested_data():
+    json_str = (
+        R'[{"0":{},"1":[],"2":{}},{"1":[[""],[]],"2":{"2":""}},'
+        R'{"0":{"a":"1"},"2":{"0":"W&RR=+I","1":""}}]'
+    )
+    df = cudf.read_json(
+        StringIO(json_str), engine="cudf_experimental", orient="records"
+    )
+    pdf = pd.read_json(StringIO(json_str), orient="records")
+
+    assert df.to_arrow().equals(pa.Table.from_pandas(pdf))
```

Comment:
This is a very roundabout way to compare these (also, arrow's equals is a bit broken).
```suggestion
    assert_eq(df, pdf)
```

Reply:
Yeah, this is unfortunate because pandas does not enforce a common schema across objects in different rows of the same column. For instance, on one row you can have: `{a:2}` and on another row `{b:4}`, whereas we would like to get `{a:2,b:null} and {a:null,b:4}. 
Hence we had to take the `pyarrow` route. `pyarrow` on the other hand, has the issue that it does not expose a reliable behaviour of ordering child columns. @galipremsagar was kind enough to propose an alternative solution that would work despite the inconsistent ordering of child columns in `pyarrow`.